### PR TITLE
Add sample screen for connectedAndInstalledNodes function

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Add sample screen in wear sample app for `connectedAndInstalledNodes` function.

Screenshot_20240123_105045

#### WHY

To demo the functionality.

#### Checklist :clipboard:
- [N/A] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [N/A] Update metalava's signature text files

### CODE REVIEW

File: datalayer/sample/phone/src/main/java/com/google/android/horologist/datalayer/sample/screens/nodeslistener/NodesListenerScreen.kt [hunk 1]
```diff
@@ -71,7 +71,7 @@ fun NodesListenerScreen(
         item {
             Text(
                 text = stringResource(id = R.string.nodes_listener_screen_header),
-                modifier = Modifier.padding(vertical = 10.dp),
+                modifier = Modifier.padding(10.dp),
```
Comment (yschimke) [inline datalayer/sample/phone/src/main/java/com/google/android/horologist/datalayer/sample/screens/nodeslistener/NodesListenerScreen.kt:74]: Why custom padding? Should `ListHeader {.Text() }` suffice?

**Review by yschimke — APPROVED** [review]

File: datalayer/sample/phone/src/main/java/com/google/android/horologist/datalayer/sample/screens/nodeslistener/NodesListenerScreen.kt (hunk 1 above)
Comment (luizgrp) [reply datalayer/sample/phone/src/main/java/com/google/android/horologist/datalayer/sample/screens/nodeslistener/NodesListenerScreen.kt:74]: This line is in the phone sample so there is no ListHeader as far as I know, but will address the one on the watch sample

Comment (luizgrp) [reply datalayer/sample/phone/src/main/java/com/google/android/horologist/datalayer/sample/screens/nodeslistener/NodesListenerScreen.kt:74]: For wear we have `Title` in Horologist compose material, which doesn't have a hard coded height as `ListHeader` does, which seems to be obsolete, see https://issuetracker.google.com/251166127.

So I'd go with Title + padding, which doesn't change much from what's already here.

Comment (yschimke) [reply datalayer/sample/phone/src/main/java/com/google/android/horologist/datalayer/sample/screens/nodeslistener/NodesListenerScreen.kt:74]: Ignore, was thinking Wear.

**Pull request merged**